django/db/migrations/migration.py

```diff
@@ -184,15 +183,25 @@ def suggest_name(self):
         name to avoid VCS conflicts if possible.
         """
         name = None
-        if len(self.operations) == 1:
-            name = self.operations[0].migration_name_fragment
-        elif (
-            len(self.operations) > 1 and
-            all(isinstance(o, operations.CreateModel) for o in self.operations)
-        ):
-            name = '_'.join(sorted(o.migration_name_fragment for o in self.operations))
-        if name is None:
-            name = 'initial' if self.initial else 'auto_%s' % get_migration_name_timestamp()
+        if self.initial:
+            return "initial"
+
+        raw_fragments = (op.migration_name_fragment for op in self.operations)
+        fragments = [name for name in raw_fragments]
+
+        migration_name_timestamp = get_migration_name_timestamp()
+        if not fragments or None in fragments:
+            return f"auto_{migration_name_timestamp}"
+        name = fragments[0]
+        left_out = len(fragments) - 1
+        for fragment in fragments[1:]:
+            new_name = name + f"_{fragment}"
+            if len(new_name) > 52:
+                break
+            name = new_name
+            left_out -= 1
+        if(left_out > 0):
+            name = name + "_and_more"
```

Comment:
`left_out` is not necessary anymore:
```suggestion
        name = fragments[0]
        for fragment in fragments[1:]:
            new_name = f'{name}_{fragment}'
            if len(new_name) > 52:
                name = f'{name}_and_more'
                break
            name = new_name
```

Reply:
I used it as a way to check for the need of `_and_more`. So in order to check for that I feel this is required. Is there any other optimized way to check for that which I am missing?

Reply:
My comment contains a suggestion.